### ISSUE TITLE
Wrong table list to ANSI join transformation when tables are qualified and aliased

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12306]
This transformation works:

```sql
-- input
select a.*, b.* from ta as a, tb as b where a.id = b.id

-- output
select
  A.*,
  B.*
from TA A
  join TB B
    on A.ID = B.ID
```

But this one doesn't (`ta` is qualified as `s.ta`):

```sql
-- input
select a.*, b.* from s.ta as a, tb as b where a.id = b.id

-- output
select
  A.*,
  B.*
from S.TA A
  cross join TB B
where A.ID = B.ID
```

**Comment by lukaseder** [comment]
The problem doesn't happen with generated code, only with parsed SQL. The problem is that `TableAlias.getQualifiedName()` isn't correct:

```java
System.out.println(table(name("S", "T")).as("A").getQualifiedName());
```

Produces

```
"S"."A"
```

I've created a separate issue to track this: https://github.com/jOOQ/jOOQ/issues/12307

**Comment by lukaseder** [comment]
Adding this override to `TableAlias` seems to fix this issue, but it might introduce other regressions:

```java

    @Override
    public Name getQualifiedName() {
        return super.getUnqualifiedName();
    }
```

Will investigate more thoroughly next week.

**Comment by lukaseder** [comment]
This seems to work and not introduce any regressions!

Fixed in jOOQ 3.16.0 and 3.15.2 (#12310)
